src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -460,5 +471,12 @@ public static string[] AppendPath(string[] path, string pathSegment)
 
             return newPath;
         }
+
+        private static T UnwrapTaskOfT<T>(Task<T> task)
+        {
+            return task.Result;
```

Comment:
.Result can lead to threadpool exhausting. usually the better way is using task.GetAwaiter().GetResult() (it's already this way in the master and probably the root issue of this PR)

Reply:
@OneCyrus Thanks, didn't know that. I've updated the PR.